### ISSUE TITLE
Using EditorVCSInterface from GDScript triggers an ERR_FAIL_COND

### ISSUE THREAD

**Issue by ahrbe1** [godotengine/godot#52195]
### Godot version

3.4.beta4

### System information

Windows 10; Ubuntu 20.04

### Issue description

I implemented a trivial subclass of EditorVCSInterface in GDScript. The engine detects it, but when clicking the VCS 'Initialize' button, an internal error is printed to the console. See attached project for a sample test-case. Tested on both Windows 10 and Ubuntu 20.04.

Error text:
`
core/object.cpp:998 - Condition "script_instance != nullptr || !script.is_null()" is true.
`

Due to a failing `ERR_FAIL_COND` near line 870 in the current repo (Line 998 in 3.4beta4):

**PermaLink:**
https://github.com/godotengine/godot/blob/e0355d5db67bcdc5b1d3f6f23793b3361b5ae88d/core/object/object.cpp#L870

This appears to be caused by **godot/editor/plugins/version_control_editor_plugin.cpp** near line 110 in the current repo (or near line 120 in 3.4beta4)
```
104	EditorVCSInterface *vcs_interface = memnew(EditorVCSInterface);
105	ScriptInstance *addon_script_instance = script->instance_create(vcs_interface);
106
107	ERR_FAIL_COND_MSG(!addon_script_instance, "Failed to create addon script instance.");
108
109	// The addon is attached as a script to the VCS interface as a proxy end-point
110	vcs_interface->set_script_and_instance(script, addon_script_instance);                 // <-- HERE
```

**PermaLink:**
https://github.com/godotengine/godot/blob/7368004869671d1ada14d0628a11a5ba536147af/editor/plugins/version_control_editor_plugin.cpp#L110

It looks to me like the script instance is already set by `script->instance_create(vcs_interface)`, and that the `set_script_and_instance` line might(?) be redundant.

### Steps to reproduce

1. Load up the sample project
2. Project -> Version Control -> Set Up Version Control -> Initialize
3. Error is printed to the console

### Minimal reproduction project

[TestSCM.zip](https://github.com/godotengine/godot/files/7071202/TestSCM.zip)

**Comment by Calinou** [comment]
cc @ChronicallySerious

**Comment by ChronicallySerious** [comment]
The error does seem to suggest what the OP suspects in the issue description. However, this use case has not been tested thoroughly before so bugs like these are there. The VCS interface was initially intended to be used in GDNative only.

However, this use case is quite reasonable in my opinion. Will get back once I am able to get some testing done on this.

**Comment by akien-mga** [comment]
Fixed by #52217.
